test/letters/actions/letters.unit.spec.js

```diff
@@ -1,86 +1,495 @@
+    const dispatch = sinon.spy();
+    thunk(dispatch, getState)
+      .then(() => {
+        const action = dispatch.secondCall.args[0];
+        expect(action.type).to.equal(SAVE_ADDRESS_FAILURE);
+      }).then(done, done);
+  });
+});
 
-        done();
-      } catch (error) {
-        done(error);
-      }
+describe('getLettersList', () => {
+  beforeEach(setup);
+  afterEach(teardown);
+
+  const lettersResponse = {
+    data: {
+      attributes: {
+        letters: [
+          { name: 'Proof of Service Letter', letterType: 'proof_of_service' },
+          { name: 'Civil Service Preference Letter', letterType: 'civil_service' }
+        ],
+        fullName: 'Mark Webb'
+      },
+      id: null,
+      type: 'evss_letters_letters_response'
+    }
+  };
+
+  it('dispatches GET_LETTERS_SUCCESS when GET succeeds', (done) => {
+    global.fetch.returns(Promise.resolve({
+      headers: { get: () => 'application/json' },
+      ok: true,
+      json: () => Promise.resolve(lettersResponse)
+    }));
+    const thunk = getLetterList();
+    const dispatch = sinon.spy();
+    thunk(dispatch, getState)
+      .then(() => {
+        const action = dispatch.firstCall.args[0];
+        expect(action.type).to.equal(GET_LETTERS_SUCCESS);
+        expect(action.data).to.eql(lettersResponse);
+      }).then(done, done);
+  });
+
+  it('dispatches GET_LETTERS_FAILURE when GET fails with generic error', (done) => {
+    global.fetch.returns(Promise.reject(new Error('something went wrong')));
+    const thunk = getLetterList();
+    const dispatch = sinon.spy();
+    thunk(dispatch, getState)
+      .then(() => {
+        const action = dispatch.firstCall.args[0];
+        expect(action.type).to.equal(GET_LETTERS_FAILURE);
+      }).then(done, done);
+  });
+
+  const lettersErrors = {
+    503: BACKEND_SERVICE_ERROR,
+    504: BACKEND_SERVICE_ERROR,
+    403: BACKEND_AUTHENTICATION_ERROR,
+    502: LETTER_ELIGIBILITY_ERROR,
+    500: GET_LETTERS_FAILURE
+  };
+
+  Object.keys(lettersErrors).forEach((code) => {
+    it(`dispatches ${lettersErrors[code]} when GET fails with ${code}`, (done) => {
+      global.fetch.returns(Promise.reject({
+        errors: [{ status: `${code}` }]
+      }));
+
+      const thunk = getLetterList();
+      const dispatch = sinon.spy();
+      thunk(dispatch, getState)
+        .then(() => {
+          const action = dispatch.firstCall.args[0];
+          expect(action.type).to.equal(lettersErrors[code]);
+        }).then(done, done);
     });
+  });
+});
+
+describe('getMailingAddress', () => {
+  beforeEach(setup);
+  afterEach(teardown);
+
+  const addressResponse = {
+    data: {
+      attributes: {
+        address: {
+          type: 'DOMESTIC',
+          addressEffectiveDate: '1973-01-01T05:00:00.000+00:00',
+          addressOne: '140 Rock Creek Church Rd NW',
+          addressTwo: '',
+          addressThree: '',
+          city: 'Washington',
+          stateCode: 'DC',
```

Comment:
🤔 :man_shrugging:

Reply:
Yeah, I stole this from the `vets-api` Swagger